#28 implement buyer personas overview as a carousel

umbrellait-pavel-bukovskiy opened the pull request:
Changes:
- final-feedback.component.html is divided into 2 components: tabs and carousel
- implemented the carousel view of final-feedback
- improved final-feedback.component.ts and feedback_section.component.ts to provide data from the builder
- improved core styles for more responsibility
- added owl-carousel dependency

umbrellait-pavel-bukovskiy commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Implement buyer personas overview as a carousel](https://trello.com/c/a0k4wp8R/3-implement-buyer-personas-overview-as-a-carousel)

integry commented:
Looks good. Would it be a big hassle to change the navigation circles with persona pictures (and make them a bit bigger)?

Also, I would like to make some visual changes to the carousel slide itself, as it should fit into a screen more easily. It would pretty much mean significantly reducing the gaps between lines and font sizes. Ideally it would look similar to the image that's attached to the original issue.

Please also make all slides the same height. Probably it makes the most sense to line them according to the tallest slide.

integry commented:
Let's keep the changes to carousel only for now. As for the persona picture size, feel free to use what looks best, we can adjust later if needed.

integry commented on src/app/module-viewer/riverside-step-template/templates/module-result/module-result.component.ts:
```diff
@@ -22,12 +23,12 @@ export class ModuleResultComponent extends TemplateComponent {
-      this.contentData.options.map(option => {
-        Object.assign(moduleOptions, { [option.key]: option.value });
-      });
+      moduleOptions = this.contentData.options.reduce((accumulator, option) => {
```
To avoid assigning via `let` you can do `const moduleOptions = (this.contentData.options || []).reduce...`

integry commented:
![image](https://user-images.githubusercontent.com/829273/74652995-ed33f580-518f-11ea-96a0-6ca3cef69fb3.png)

I think `::ng-deep` should be used instead of `/deep/`

umbrellait-pavel-bukovskiy commented:
I fixed /deep/ and improved code style. Also, I added scrolling to top on slide change and style improvements of slide content to more similarity with attached to the issue mock

integry commented:
Looks great. Just one more thing, I'm not sure if I described it clearly enough. Is it trivially doable to replace this part with persona pictures?

![image](https://user-images.githubusercontent.com/829273/74654369-d216b500-5192-11ea-965f-930605e30502.png)

umbrellait-pavel-bukovskiy commented:
> Looks great. Just one more thing, I'm not sure if I described it clearly enough. Is it trivially doable to replace this part with persona pictures?
> 
> ![image](https://user-images.githubusercontent.com/829273/74654369-d216b500-5192-11ea-965f-930605e30502.png)

I'll try to do this, but I think the size of personas picture should be about 60px size

integry commented:
That would be fine, we don't need them to be super-clearly identifiable from those small pictures.

integry commented:
Looks good, but there's no need to keep the navigation circles. In addition, can you please change the cursor to pointer and add border to the image of the active persona?

![image](https://user-images.githubusercontent.com/829273/74663590-bae0c300-51a4-11ea-9e56-9898ae486efe.png)

integry commented:
I made that change myself as I'm trying to clear the current PRs before I leave tomorrow.

integry commented:
Great job with this one, merging it.
